src/mbgl/text/glyph_atlas.cpp

```diff
@@ -64,16 +65,8 @@ bool GlyphAtlas::hasGlyphRanges(const FontStack& fontStack, const GlyphRangeSet&
 }
 
 util::exclusive<GlyphSet> GlyphAtlas::getGlyphSet(const FontStack& fontStack) {
-    auto lock = std::make_unique<std::lock_guard<std::mutex>>(glyphSetsMutex);
-
-    auto it = glyphSets.find(fontStack);
-    if (it == glyphSets.end()) {
-        it = glyphSets.emplace(fontStack, std::make_unique<GlyphSet>()).first;
-    }
-
-    // FIXME: We lock all GlyphSets, but what we should
-    // really do is lock only the one we are returning.
```

Comment:
No, but I don't think it'll ever be fixed in the way described here. Instead, we'll do #667.